### PR TITLE
qs: add undefined to ParsedQs union type

### PR DESCRIPTION
**Pull request by pscanf**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by pscanf** [comment]
As discussed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43434#discussion_r422916691 (@OliverJAsh)

File: types/qs/qs-tests.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ qs.parse('a=b&c=d', { delimiter: '&' });
 () => {
     let obj = qs.parse('a=z&b[c]=z&d=z&d=z&e[][f]=z');
     obj; // $ExpectType ParsedQs
-    obj.a; // $ExpectType string | ParsedQs | string[] | ParsedQs[]
+    obj.a; // $ExpectType string | ParsedQs | string[] | ParsedQs[] | undefined
```
Comment (OliverJAsh) [inline types/qs/qs-tests.ts:13]: Maybe add a test to cover your specific use case, so it's clearer why we want to include `undefined`?

Comment (OliverJAsh) [reply types/qs/qs-tests.ts:13]: Something like:

```ts
declare const myQuery: { a: string; b?: string }
const myQueryCopy: ParsedQs = myQuery;
```

Comment (pscanf) [reply types/qs/qs-tests.ts:13]: I added the interface extension test that should cover the same thing, but I can also add this one no problem.

Comment (pscanf) [reply types/qs/qs-tests.ts:13]: Added in df7ec738a73d787712d9b71013720c513a29486b

Comment (kpping) [reply types/qs/qs-tests.ts:13]: I have already PR to add undefined to [ParsedQS](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44617#pullrequestreview-408767023)

Comment (pscanf) [reply types/qs/qs-tests.ts:13]: Ah, sorry, completely missed it. I can close this one then, maybe you can just port over the couple of test cases I added to cover the use case.

Comment (kpping) [reply types/qs/qs-tests.ts:13]: No problem. It's completely fine. I have add your test case into the PR #44617 with this commit 7c85f0aa0127835782d67ebfcf2a86d6979a0638

Comment (pscanf) [reply types/qs/qs-tests.ts:13]: Perfect, thanks, then I'll close this one.

**Comment by pscanf** [comment]
Duplicate of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44617 which I had not seen (sorry).

**Pull request closed without merging**